packages/cli/src/api/extractors/babel.ts

```diff
@@ -1,41 +1,62 @@
-import {DEFAULT_EXTENSIONS, transformFileSync} from "@babel/core"
+import {DEFAULT_EXTENSIONS, transformAsync, } from "@babel/core"
 
 import type {ExtractPluginOpts} from "@lingui/babel-plugin-extract-messages"
 import linguiExtractMessages from "@lingui/babel-plugin-extract-messages"
 
-import {BabelOptions, ExtractorType} from "."
-import {projectType} from "../detect"
+import {ExtractorType} from "."
+import {ParserPlugin} from "@babel/parser"
 
 const babelRe = new RegExp(
-	"\\.(" + [...DEFAULT_EXTENSIONS, ".ts", ".tsx"].map(ext => ext.slice(1)).join("|") + ")$",
-	"i"
+  "\\.(" + [...DEFAULT_EXTENSIONS, ".ts", ".mts", ".cts", ".tsx"].map(ext => ext.slice(1)).join("|") + ")$",
+  "i"
 )
 
 const extractor: ExtractorType = {
   match(filename) {
     return babelRe.test(filename)
   },
 
-  extract(filename, onMessageExtracted, options = {}) {
-    const { babelOptions: _babelOptions = {} } = options
-    const { plugins = [], ...babelOptions } = _babelOptions
-    const frameworkOptions: BabelOptions = {}
+  async extract(filename, code, onMessageExtracted, options = {}) {
+    const parserPlugins: ParserPlugin[] = [
+      // https://babeljs.io/docs/en/babel-parser#latest-ecmascript-features
+      ["decorators", {
+        decoratorsBeforeExport: options?.parserOptions?.decoratorsBeforeExport || true
+      }]
+    ]
```

Comment:
By default, babel parser can understand syntax up to es2022, no additional parser plugins needed. 
We add here support for decorators because they are extremely popular in TS projects.